pkg/ottl/ottlfuncs/README.md

```diff
@@ -1074,6 +1075,34 @@ Examples:
 
 - `Hours(Duration("1h"))`
 
+### Index
+
+`Index(source, value)`
+
+The `Index` Converter returns the index of the first occurrence of the `value` in the `source`, or -1 if not found.
+
+**Supported Source Types:**
+
+`source` can be `pcommon.Slice`, `pcommon.Value`, or following native Go slices:
+- `[]any` - slice with mixed types
+- `[]string` - string slice
+- `[]int`, `[]int16`, `[]int32`, `[]int64` - integer slices
+- `[]uint`, `[]uint16`, `[]uint32`, `[]uint64` - unsigned integer slices
+- `[]float32`, `[]float64` - floating-point slices
+- `[]bool` - boolean slice
+
+**Type requirements:**
+- For numeric types, the search `value` should be `int64` or `float64`
+- String and boolean comparisons require exact type matches
```

Comment:
```suggestion
`source` is a string or a slice of any type described in the OTTL comparison rules.

`value` is the content whose index is being searched for in the `source`.
```

Reply:
Yup, make sense to abstract and link to comparison rule. Thanks.